CMakeLists.txt

```diff
@@ -22,31 +22,93 @@ find_package(rclcpp REQUIRED)
 find_package(shisen_interfaces REQUIRED)
 find_package(sensor_msgs REQUIRED)
 find_package(std_msgs REQUIRED)
+find_package(Protobuf CONFIG REQUIRED)
+message(STATUS "Using protobuf ${Protobuf_VERSION}")
 
-install(DIRECTORY "include" DESTINATION ".")
+# Find gRPC installation
+# Looks for gRPCConfig.cmake file installed by gRPC's cmake installation.
+find_package(gRPC CONFIG REQUIRED)
+message(STATUS "Using gRPC ${gRPC_VERSION}")
 
 add_library(${PROJECT_NAME} SHARED
   "src/${PROJECT_NAME}/camera/node/camera_node.cpp"
   "src/${PROJECT_NAME}/camera/provider/camera_config_provider.cpp"
   "src/${PROJECT_NAME}/camera/provider/image_provider.cpp"
+  "src/${PROJECT_NAME}/config/grpc/config.cpp"
+  "src/${PROJECT_NAME}/config/grpc/call_data_base.cpp"
+  "src/${PROJECT_NAME}/config/grpc/call_data_get_capture_setting.cpp"
+  "src/${PROJECT_NAME}/config/grpc/call_data_get_image.cpp"
+  "src/${PROJECT_NAME}/config/grpc/call_data_save_capture_setting.cpp"
+  "src/${PROJECT_NAME}/config/grpc/call_data_set_capture_setting.cpp"
+  "src/${PROJECT_NAME}/config/utils/config.cpp"
   "src/${PROJECT_NAME}/utility/capture_setting.cpp"
   "src/${PROJECT_NAME}/utility/interface.cpp"
   "src/${PROJECT_NAME}/node/shisen_cpp_node.cpp"
   "src/${PROJECT_NAME}/viewer/consumer/image_consumer.cpp"
-  "src/${PROJECT_NAME}/viewer/node/viewer_node.cpp")
+  "src/${PROJECT_NAME}/viewer/node/viewer_node.cpp"
+)
+
+add_library(${PROJECT_NAME}_exported SHARED
+  "src/${PROJECT_NAME}/camera/node/camera_node.cpp"
+  "src/${PROJECT_NAME}/camera/provider/camera_config_provider.cpp"
+  "src/${PROJECT_NAME}/camera/provider/image_provider.cpp"
+  "src/${PROJECT_NAME}/utility/capture_setting.cpp"
+  "src/${PROJECT_NAME}/utility/interface.cpp"
+  "src/${PROJECT_NAME}/viewer/consumer/image_consumer.cpp"
+  "src/${PROJECT_NAME}/viewer/node/viewer_node.cpp"
+)
```

Comment:
Why is this being separated?

Reply:
It was separated because if the grpc file is also exported, it will cause a conflict for other packages that use grpc too and also call this package (in this case ```ninshiki_cpp```)

The error message:
```ERROR: Something is wrong with flag 'grpc_experiments' in file 'home/mbsaloka/grpc/src/core/lib/config/config_vars.cc'. One possibility: file 'home/mbsaloka/grpc/src/core/lib/config/config_vars.cc' is being linked both statically and dynamically into this executable. e.g. some files listed as srcs to a test and also listed as srcs of some shared lib deps of the same test.```

Reply:
Can you DM me the full log?